components/ws-daemon/pkg/cpulimit/cfs.go

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package cpulimit
+
+import (
+	"bufio"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
+	"golang.org/x/xerrors"
+)
+
+// CgroupCFSController controls a cgroup's CFS settings
+type CgroupCFSController string
+
+// Usage returns the cpuacct.usage value of the cgroup
+func (basePath CgroupCFSController) Usage() (usage CPUTime, err error) {
+	fn := filepath.Join(string(basePath), "cpuacct.usage")
+	fc, err := os.ReadFile(fn)
+	if err != nil {
+		return 0, xerrors.Errorf("cannot sample cpuacct.usage: %w", err)
+	}
+
+	cpuTimeInNS, err := strconv.ParseInt(strings.TrimSpace(string(fc)), 10, 64)
```

Comment:
I'm sorry if you have a plan to add handling it. We need to pay attention to zero-valued because a zero-valued cgroup value sometimes has a special meaning. For example, in `cpuacct.usage` zero means reset.
https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/6/html/resource_management_guide/sec-cpuacct

Reply:
Great point - thanks for pointing that out

Reply:
Maybe the code of runc is helpful.
https://github.com/opencontainers/runc/blob/ede712786ce84d8e9953b694149428fc9dbe0d74/libcontainer/cgroups/fscommon/utils.go#L101

Reply:
Thanks - the pointers were helpful indeed :)

Reply:
@csweichel just following up on above ^
seems like you need  to improve handling here?
should probably use `readUint64` function